### PR TITLE
Fixes for HDR support.

### PR DESCRIPTION
**Pull request by kleinerm**
This pull contains bug fixes and improvements for HDR support to make HDR actually usable for desktop applications, which will generally run under X11 and not directly on a VT DRM framebuffer.

1. Fix the EDID parsing wrt. HDR display capabilities.

2. Make HDR work with X11 RandR leased video outputs in direct display mode.

Please consider these fixes for a release soon. Thanks.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JacobHeAMD** [comment]
ok to test

**Comment by JacobHeAMD** [comment]
test this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit fc1861276fd8e856ef3389dc8fa7b521446f7e80
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>LLPC: dev faaf16cb0492c67eb56caa2d9a1d479d1764104c</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 7b14353794b408b8d7fd87ae9a9f4f553e6d4782</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103142/103142 (100.0%)</li>
<li>Failed: 0/103142 (0.0%)</li>
</ul></details>


</ul></details>

File: src/core/os/amdgpu/amdgpuScreen.cpp [hunk 1]
```diff
@@ -61,13 +61,15 @@ Screen::Screen(
...
+    // Restore to SDR, unless already in SDR mode.
+    if (m_userColorGamut.metadata.eotf == HDMI_EOTF_TRADITIONAL_GAMMA_SDR)
+        return;
 
     m_userColorGamut.metadataType          = HDMI_STATIC_METADATA_TYPE1;
     m_userColorGamut.metadata.eotf         = HDMI_EOTF_TRADITIONAL_GAMMA_SDR;
     m_userColorGamut.metadata.metadataType = HDMI_STATIC_METADATA_TYPE1;
 
-    static_cast<Device*>(m_pDevice)->SetHdrMetaData(m_connectorId, &m_userColorGamut);
+    static_cast<Device*>(m_pDevice)->SetHdrMetaData(m_drmMasterFd, m_connectorId, &m_userColorGamut);
```
Comment (brianbAMD) [inline src/core/os/amdgpu/amdgpuScreen.cpp:72]: Our coding standards don't allow early return statements.

**Comment by kleinerm** [comment]
Updated. Thanks for the quick review.

**Comment by kleinerm** [comment]
Btw. while testing the recent Vulkan driver on Windows-10, i also found a HDR EDID parsing bug there. In the returned VkHdrMetadataEXT structure, the subfield maxLuminance contains what should be in subfield maxFrameAverageLightLevel, whereas the subfield maxFrameAverageLightLevel contains zero, instead of the EDID provided data.

Maybe this could be fixed or forwarded internally, assuming this is the same team for PAL on Windows?

E.g., for my Samsung C27HG70 HDR monitor:

WRONG on Windows:

Vulkan driver/gpu: AMD Radeon(TM) RX Vega 11 Graphics
Display native HDR properties as queried from monitor on AMD Vulkan on Windows-10:
Display Supports control of HDR local dimming: No
Display Gamut  R: [0.676800, 0.316400]
Display Gamut  G: [0.275400, 0.661100]
Display Gamut  B: [0.152300, 0.061500]
Display Gamut WP: [0.312500, 0.329100]
Display minLuminance: 0.049100 nits
Display maxLuminance: 351.000000 nits
Content maxFrameAverageLightLevel: 0.000000 nits
Content maxContentLightLevel: 0.000000 nits

RIGHT on Linux:

Display native HDR properties as queried from monitor from bug-fixed amdvlk:
Display Supports control of HDR local dimming: No
Display Gamut  R: [0.676700, 0.316400]
Display Gamut  G: [0.275300, 0.661100]
Display Gamut  B: [0.152300, 0.061500]
Display Gamut WP: [0.313400, 0.329100]
Display minLuminance: 0.049000 nits
Display maxLuminance: 603.000000 nits
Content maxFrameAverageLightLevel: 351.000000 nits
Content maxContentLightLevel: 0.000000 nits

Thanks.

**Comment by JacobHeAMD** [comment]
test this please

**Comment by JacobHeAMD** [comment]
@kleinerm , thanks for your information. I'll forward it to proper team to have a look.

**Pull request merged**